.github/workflows/testing-macos.yml

```diff
@@ -0,0 +1,57 @@
+name: build and test on macOS
+
+on:
+  push:
+    branches: [ master, devel ]
+  pull_request:
+    branches: [ master, devel ]
+
+jobs:
+  debug:
+
+    strategy:
+      matrix:
+        os: [macos-14, macos-13, macos-12]
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: recursive
+
+      - run: brew install bash gnu-time bats-core
```

Comment:
You probably should install `git` too, otherwise `update_version.sh` doesn't work correctly.

Reply:
git has already pre-installed. 

`actions/checkout@v4` only fetch the latest one commit by default. no tags are fetched. 

add `fetch-tags: true` would resolve the problem.